src/AudioGeneratorMP3.cpp

```diff
@@ -48,28 +54,27 @@ AudioGeneratorMP3::AudioGeneratorMP3(void *space, int size)
 AudioGeneratorMP3::~AudioGeneratorMP3()
 {
   if (!preallocateSpace) {
-    free(buff);
-    free(synth);
-    free(frame);
-    free(stream);
+    if (buff) free(buff);
```

Comment:
No need for this check.  free(NULL) is perfectly legal and guaranteed to do nothing.

Reply:
Good to know, I'll roll this (and related checks) back out.